src/components/radio-button/radio-button-group/radio-button-group.test.tsx

```diff
@@ -0,0 +1,292 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { RadioButton, RadioButtonGroup } from "..";
+import CarbonProvider from "../../carbon-provider";
+import Logger from "../../../__internal__/utils/logger";
+import { mockMatchMedia } from "../../../__spec_helper__/__internal__/test-utils";
+
+test("logs a deprecation warning for uncontrolled behaviour", () => {
+  const loggerSpy = jest
+    .spyOn(Logger, "deprecate")
+    .mockImplementation(() => {});
+  render(
+    <RadioButtonGroup name="group">
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Radio Button` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+test("renders with `RadioButton` children", () => {
+  render(
+    <RadioButtonGroup name="group" onChange={() => {}}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const radioButton = screen.getByRole("radio", { name: "Radio Button 1" });
+
+  expect(radioButton).toBeInTheDocument();
+  expect(screen.getByTestId("radio-svg")).toBeVisible();
+  expect(screen.getByText("Radio Button 1")).toBeVisible();
+});
+
+test("renders with non-`RadioButton` children when passed as an array", () => {
+  render(
+    <RadioButtonGroup name="group" onChange={() => {}}>
+      {[
+        <RadioButton key="radio1" value="radio1" label="Radio Button 1" />,
+        null,
+        undefined,
+        "foo",
+        <RadioButton key="radio2" value="radio2" label="Radio Button 2" />,
+      ]}
+    </RadioButtonGroup>
+  );
+
+  expect(screen.getAllByRole("radio")).toHaveLength(2);
+  expect(screen.getByText("foo")).toBeVisible();
+});
+
+test("renders fieldset with provided `legend`", () => {
+  render(
+    <RadioButtonGroup
+      name="group"
+      legend="Radio Group Legend"
+      onChange={() => {}}
+    >
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const fieldset = screen.getByRole("group", { name: "Radio Group Legend" });
```

Comment:
**praise**: great job checking the ARIA roles! Your method of targeting elements mirrors how an assistive technology user would inspect the page 👍🏼